python/hail/methods/statgen.py

```diff
@@ -694,21 +699,36 @@ def pc_relate(dataset, k, maf, block_size=512, min_kinship=-float("inf"), statis
         kinship and identity-by-descent zero, one, and two.
 
         The fields of the resulting :class:`.Table` entries are of types:
-        `i`: `String`, `j`: `String`, `kin`: `Double`, `k2`: `Double`,
+        `i`: ``ds.colkey_schema``, `j`: ``ds.colkey_schema``, `kin`: `Double`, `k2`: `Double`,
         `k1`: `Double`, `k0`: `Double`. The table is keyed by `i` and `j`.
     """
-    dataset = require_biallelic(dataset, 'pc_relate')
+    ds = require_biallelic(ds, 'pc_relate')
+    ds = require_unique_samples(ds, 'pc_relate')
+    ds = ds.annotate_rows(total_alt_alleles=agg.sum(ds.GT.num_alt_alleles()),
+                          n_called=agg.count_where(functions.is_defined(ds.GT)))
+
+    mean_imputed_gt = functions.bind(
```

Comment:
bind doesn't help since mean_gt only occurs once on right, and the bound expression will still end up being computed once per call (entry) rather than per row.  Instead:
```
ds = ds.annotate_rows(mean_gt = ds.total_alt_alleles / ds.n_called)
mean_imputed_gt = or_else(ds.GT.tofloat64(), mean_gt)
```